include/nlohmann/detail/diagnostics_t.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include <nlohmann/detail/value_t.hpp>
+#include <nlohmann/detail/string_escape.hpp>
+
+namespace nlohmann
+{
+namespace detail
+{
+
+template<typename BasicJsonType>
+class diagnostics_t
+{
+  public:
+    diagnostics_t() noexcept = default;
+    diagnostics_t(const BasicJsonType& j) noexcept
+        : m_j(&j)
+    {}
+
+    std::string diagnostics() const
+    {
+#if JSON_DIAGNOSTICS
+        if (m_j == nullptr)
+        {
+            return "";
+        }
+
+        std::vector<std::string> tokens;
+        for (const auto* current = m_j; current->m_parent != nullptr; current = current->m_parent)
+        {
+            switch (current->m_parent->type())
+            {
+                case value_t::array:
+                {
+                    for (std::size_t i = 0; i < current->m_parent->m_value.array->size(); ++i)
+                    {
+                        if (current->m_parent->m_value.array->operator[](i) == *current)
+                        {
+                            tokens.emplace_back(std::to_string(i));
+                            continue;
```

Comment:
`continue` is meaningless here, did you mean `break` to exit the loop?

Reply:
Good catch!

Reply:
Fixed in d6ff059a901fc69163d9a42e15fcef87278fb23d.